### PR TITLE
Add terradoc

### PR DESCRIPTION
**Pull request by kassianh**
Recreated README with terradoc, checked documentation for errors or inconsistencies.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -105,10 +105,16 @@ help:
...
+
 # Define helper functions
 DOCKER_FLAGS   += ${DOCKER_RUN_FLAGS}
 DOCKER_RUN_CMD  = docker run ${DOCKER_FLAGS} ${BUILD_TOOLS_DOCKER_IMAGE}
 
 quiet-command = $(if ${V},${1},$(if ${2},@echo ${2} && ${1}, @${1}))
 docker-run    = $(call quiet-command,${DOCKER_RUN_CMD} ${1} | cat,"${YELLOW}[DOCKER RUN] ${GREEN}${1}${RESET}")
-rm-command    = $(call quiet-command,rm -rf ${1},"${YELLOW}[CLEAN] ${GREEN}${1}${RESET}")
+go-test       = $(call quiet-command,${DOCKER_RUN_CMD} go test -v -count 1 -timeout 45m -parallel 128 ${1} | cat,"${YELLOW}[TEST] ${GREEN}${1}${RESET}")
+rm-command    = $(call quiet-command,rm -rf ${1},"${YELLOW}[CLEAN] ${GREEN}${1}${RESET}")
```
Comment (mariux) [inline Makefile:120]: please fix EOL at EOF in all PRs and maybe fix you editor settings ;)

**Review by mariux — CHANGES_REQUESTED** [review]

File: README.tfdoc.hcl [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+        variable "module_enabled" {
+          type        = bool
+          default     = true
+          description = <<-END
+            Specifies whether resources in the module will be created.
+          END
+        }
+
+        variable "module_depends_on" {
+          type           = list(any)
```
Comment (mariux) [inline README.tfdoc.hcl:94]:
```suggestion
          type           = any
```

File: README.tfdoc.hcl [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+        variable "role" {
+          type = any
+          readme_type = <<-END
+            list(object({
+            role           = string
+            special_group  = string
+            group_by_email = string
+            user_by_email  = string
+          }))
+          END
```
Comment (mariux) [inline README.tfdoc.hcl:261]: complex types are just `list(type)`  we do not expand objects as we explain eahc attribute anyway.

is role a list of roles? looks like the naming is bad but can be fixed at another time..



```suggestion
          readme_type = <<-END
            list(object({
            role           = string
            special_group  = string
            group_by_email = string
            user_by_email  = string
          }))
          END
```

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+          description = <<-END
+            If set to true, delete all the tables in the dataset when destroying the resource; otherwise, destroying the resource will fail if tables are present.
+          END
+        }
+
+        variable "authoritative" {
+          type        = bool
+          default     = true
+          description = <<-END
+            (Optional) google bigquery dataset iam choice whether 'authoritative == true' use google_bigquery_iam_binding, 'authoritative == false' use google_bigquery_iam_member
```
Comment (mariux) [inline README.tfdoc.hcl:297]: this text is outdated... please replace whenever you see this.

```suggestion
            Whether to exclusively set (authoritative mode) or add (non-authoritative/additive mode) members to the role.
```

File: README.tfdoc.hcl [hunk 5]
```diff
@@ -0,0 +1,531 @@
...
+          description = <<-END
+            A list of IAM access to apply to the created secret.
+          END
+
+          attribute "role" {
+            required    = true
+            type        = string
+            description = <<-END
+            The role that should be applied. Note that custom roles must be of the format [projects|organizations]/{parent-name}/roles/{role-name}.
+          END
```
Comment (mariux) [inline README.tfdoc.hcl:314]: wrong indent..

```suggestion
            description = <<-END
              The role that should be applied. Note that custom roles must be of the format [projects|organizations]/{parent-name}/roles/{role-name}.
            END
```

**Review by mariux — CHANGES_REQUESTED** [review]
please fix those issues in all open pRs and future ones..

File: README.tfdoc.hcl [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+            description = <<-END
+              The ID of the project containing this table.
+            END
+          }
+
+          attribute "table_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the dataset containing this table.
```
Comment (soerenmartius) [inline README.tfdoc.hcl:239]: this seems to be mixed up with the variable below? this is the table_id and not the dataset id?

File: README.tfdoc.hcl [hunk 7]
```diff
@@ -0,0 +1,524 @@
...
+            END
+          }
+
+          attribute "dataset_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the table.
```
Comment (soerenmartius) [inline README.tfdoc.hcl:247]: I guess here we need to use the description of the variable above?

**Review by soerenmartius — DISMISSED** [review]

File: README.tfdoc.hcl [hunk 8]
```diff
@@ -0,0 +1,524 @@
...
+
+      The iam resource objects that define the access to the secret.
+    END
+  }
+
+  section {
+    title = "External Documentation"
+
+    section {
+      title   = "Google Documentation:"
```
Comment (mariux) [inline README.tfdoc.hcl:363]:
```suggestion
      title   = "Google Documentation"
```

File: README.tfdoc.hcl [hunk 9]
```diff
@@ -0,0 +1,524 @@
...
+      content = <<-END
+        - Bigquery Dataset: <https://cloud.google.com/bigquery/docs/datasets-intro>
+        - Bigquery Access Control: <https://cloud.google.com/bigquery/docs/access-control>
+      END
+    }
+
+    section {
+      title   = "Terraform Google Provider Documentation:"
```
Comment (mariux) [inline README.tfdoc.hcl:371]:
```suggestion
      title   = "Terraform Google Provider Documentation"
```

**Review by mariux — DISMISSED** [review]

**Review by i4ki — APPROVED** [review]
lgtm

**Pull request merged**